content/waku/2024-gantt.md

```diff
@@ -0,0 +1,105 @@
+
+Colour legend:
+
+- **Red**: engineering work to deliver the feature.
+- Other: test and telemetry work to ensure quality
+
+golang engineers: working on status-go/go-waku
+nim engineers: working on nwaku, may be a CC from research team too.
+
+Anything prefixed `TBC` is pending confirmation of estimate with the team.
+
+```mermaid
+gantt
+    dateFormat YYYY-MM-DD
+    axisFormat %d-%b
+    tickInterval 1week
+    weekday monday
+    
+%% Milestones overview with deliverables
+    section Store Service Upgrade
+        Store v3-beta (msg hash): crit, milestone, after storev3-br, 0
+        Store v3 (sync): crit, milestone, after storev3-r storev3-g storev3-n, 0
+        DoS protection for req-res protocols: crit, milestone, after dosreqresn dosreqresj dosreqresg, 0
+        PostgreSQL maintenance: crit, milestone, after pgsql, 0
+        (metric) Count store messages: milestone, after count-store-msg, 0
+    section Direct Message Reliability
+        (testing) direct messages: milestone, after test-direct-msg, 0
+        Review connection management: crit, milestone, after review-conn-mgmt-1 review-conn-mgmt-2, 0
+        (telemetry) Fleet logging: milestone, after telem-fleet-logging, 0
+        (telemetry) direct message reliability: milestone, after telem-d-msg-rel, 0
+        Reliability Protocol for Relay: crit, milestone, after rel-relay-g rel-relay-n, 0
+        Reliability Protocol for Resource-Restricted Clients: crit, milestone, after rel-reqres-g rel-reqres-j, 0
+        Review MVDS usage and fail path: crit, milestone, after mvds, 0
+    section Static Sharding - dedicated shards
+        (telemetry) Measure Bandwidth: milestone, after telem-bandwidth, 0
+        Sharding peer mgmt and discovery hardening: crit, milestone, after sh-peer-mgmt-j sh-peer-mgmt-n sh-peer-mgmt-g, 0
+        (testing) Custom shard impl of Communities: milestone, after test-custom-shard, 0
+    section E2e reliability protocol
+        (telemetry) Multicast message reliability: milestone, after telem-m-msg-rel, 0
+        E2e reliability protocol PoC: milestone, crit, after e2e-rel-r, 0
+        E2e reliability protocol Status integration: milestone, crit, after e2e-rel-g, 0
+
+%% Tasks
+    section golang eng 1
+        %% Estimation TBC - Prem says fine, waiting on 2nd opinion
+        TBC Review connection management: crit, review-conn-mgmt-1, 2024-05-13, 8w
+        Review MVDS usage and fail path: crit, mvds, after review-conn-mgmt-1, 6w
+        Investigation and fixing of bugs discovered during dogfooding/usage/simulations: go-bugs-1, after mvds, 8w
+    section golang eng 2
+        Reliability Protocol for Relay (go): crit, rel-relay-g, 2024-05-13, 12w
+        (testing) direct messages: test-direct-msg, after rel-relay-g, 4w
+        TBC (testing) Custom shard impl of Communities: test-custom-shard, after test-direct-msg, 4w
+        %% TBC estimate
+        TBC E2e reliability protocol Status integration: crit, e2e-rel-g, after e2e-rel-r, 10w
```

Comment:
Once the library is ready, how much work are we thinking of? @chaitanyaprem @kaichaosun